pkg/agent/svid/rotator_test.go

```diff
@@ -96,23 +96,18 @@ func TestRotator(t *testing.T) {
 				errCh <- rotator.Run(ctx)
 			}()
 
-			// Wait for the initial rotation check to finish
-			select {
-			case <-time.After(time.Minute):
-				t.Fatal("timed out waiting for rotation check to finish")
-			case <-rotationDone:
-			}
```

Comment:
I think we still need this lock-step in order for the tests to really assert the right things took place. What _is_ missing, and what fixes the race, is that we are not making sure that the rotator loop has made it back around and started waiting (via a select on r.clk.After) before we add time to the mock clock. I think adding adding a wait here for that After call fixes the race, while still keeping the right assertions in place.

```suggestion
			case <-rotationDone:
				clk.WaitForAfter(time.Minute, "waiting for rotation loop to start waiting")
```

Reply:
Thanks for this suggestion! This clears up my confusion on how two runRotation loops could complete too.

Reply:
One thing that this also made more obvious was a timing collision between `notAfter` and `checkAfter` in the "expires after startup" test case (https://github.com/spiffe/spire/pull/2738/files#diff-2629a31628d2e08dceb05fa87836bc509d5410d142e61e1e69974aae1d485156R50) which I fixed by adding one additional second to `checkAfter`